csa.js

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env node
+
+'use strict'
+
+const DEP = 0, ARR = 1, DEP_TS = 2, ARR_TS = 3
+const MAX_STATION = 100000
+let timetable = new Uint32Array(4)
+let timetableLength = 4
+
+function compute(trip) { // Crankshaft-optimizable (Node v5.0.0)
+  let earliestArr = new Uint32Array(MAX_STATION).fill(-1)
+  let inConn = []
+  let route = []
+  let station
+
+  earliestArr[trip[DEP]] = trip[DEP_TS]
+
+  for (let offset = 0 ; offset < timetableLength ; offset = offset + 4 /* [1] */) {
+    // [1]: Crankshaft doesn't support let compound assignements ('offset += 4').
+    if (
+      timetable[offset + DEP_TS] >= earliestArr[timetable[offset + DEP]] &&
+      timetable[offset + ARR_TS] <  earliestArr[timetable[offset + ARR]]
+    ) {
+      inConn[timetable[offset + ARR]] = new Uint32Array(
+        timetable.buffer, offset * Uint32Array.BYTES_PER_ELEMENT, 4
+      )
```

Comment:
Good point! I guess it would be more efficient, yes. We could also allocate `earliestArr` and `inConn` outside of `compute`. I'll give it a try.